node/packages/parser/src/ethereum/index.ts

```diff
@@ -79,6 +76,58 @@ export class TransactionParser {
       value: tx.value,
     }
 
+    const pTxWithTransfers = this.getParsedTxWithTransfers(tx, pTx, address, internalTxs)
+
+    // Add metadata and return
+    return {
+      ...pTxWithTransfers,
+      data: {
+        buyTx: getBuyTx(pTxWithTransfers),
+        sellTx: getSellTx(pTxWithTransfers),
+      },
+    }
+  }
+
+  // keep track of all individual tx components and add up the total value transferred by to/from address
+  private aggregateTransfer(
+    transfers: Array<Transfer>,
+    type: TransferType,
+    caip19: string,
+    from: string,
+    to: string,
+    value: string,
+    token?: Token
+  ): Array<Transfer> {
+    if (!new BigNumber(value).gt(0)) return transfers
+
+    const index = transfers?.findIndex((t) => t.type === type && t.caip19 === caip19 && t.from === from && t.to === to)
+    const transfer = transfers?.[index]
+
+    if (transfer) {
+      transfer.totalValue = new BigNumber(transfer.totalValue).plus(value).toString(10)
+      transfer.components.push({ value: value })
+      transfers[index] = transfer
```

Comment:
I know this is just moved code, but since we return `transfer` and use it like so `        pTx.transfers = this.aggregateTransfer(...args)`, do we actually want to mutate `transfer`?
To make it cleaner and avoid side-effects, we could avoid using references and just (deep) clone the object return a new one with all fields declaratively set (the non-changed ones could be spread from old object)

Reply:
This has been brought up before and is a valid concern. I will make a small issue ticket for this

Reply:
Noting here that we've DMed about this, and I agree with where your head is at.

I'll merge this as-is, but expect more PRs in this direction 👌